src/features/machine/PreviewWindow.js

```diff
@@ -159,32 +159,39 @@ class PreviewWindow extends Component {
     context.lineWidth = "1"
     context.strokeStyle = "lightblue"
     if (this.props.use_rect) {
-      this.moveTo_mm(context, Vertex((this.props.minX - this.props.maxX)/2.0, (this.props.minY - this.props.maxY)/2.0))
-      this.lineTo_mm(context, Vertex((this.props.maxX - this.props.minX)/2.0, (this.props.minY - this.props.maxY)/2.0))
-      this.lineTo_mm(context, Vertex((this.props.maxX - this.props.minX)/2.0, (this.props.maxY - this.props.minY)/2.0))
-      this.lineTo_mm(context, Vertex((this.props.minX - this.props.maxX)/2.0, (this.props.maxY - this.props.minY)/2.0))
-      this.lineTo_mm(context, Vertex((this.props.minX - this.props.maxX)/2.0, (this.props.minY - this.props.maxY)/2.0))
+      this.moveTo_mm(context, new Victor((this.props.minX - this.props.maxX)/2.0, (this.props.minY - this.props.maxY)/2.0))
+      this.lineTo_mm(context, new Victor((this.props.maxX - this.props.minX)/2.0, (this.props.minY - this.props.maxY)/2.0))
+      this.lineTo_mm(context, new Victor((this.props.maxX - this.props.minX)/2.0, (this.props.maxY - this.props.minY)/2.0))
+      this.lineTo_mm(context, new Victor((this.props.minX - this.props.maxX)/2.0, (this.props.maxY - this.props.minY)/2.0))
+      this.lineTo_mm(context, new Victor((this.props.minX - this.props.maxX)/2.0, (this.props.minY - this.props.maxY)/2.0))
     } else {
-      this.moveTo_mm(context, Vertex(this.props.maxRadius, 0.0))
+      this.moveTo_mm(context, new Victor(this.props.maxRadius, 0.0))
       let resolution = 128.0
       for (let i=0; i<=resolution; i++) {
         let angle = Math.PI * 2.0 / resolution * i
-        this.lineTo_mm(context, Vertex(this.props.maxRadius * Math.cos(angle),
+        this.lineTo_mm(context, new Victor(this.props.maxRadius * Math.cos(angle),
                                        this.props.maxRadius * Math.sin(angle)))
       }
     }
     context.stroke()
 
     if (this.props.vertices && this.props.vertices.length > 0) {
-      let drawing_vertices = this.slice_vertices(this.props.vertices, this.props.sliderValue)
+      let sliderRange = this.sliderVertexRange(this.props.vertices, this.props.sliderValue)
+      let drawing_vertices = this.props.vertices.slice(sliderRange[0], sliderRange[1] + 1)
 
       // Draw the background vertices
       if (this.props.sliderValue !== 0) {
         context.beginPath()
         context.lineWidth = this.mmToPixelsScale()
-        context.strokeStyle = "rgba(204, 204, 204, 0.35)"
+        context.strokeStyle = Color('#6E6E00')
```

Comment:
I was skeptical, but I really like how this ends up looking. It still gives me a sense of start to end, but now I also don't have to remember which points are in the past vs. future.